force-app/main/default/lwc/attendance/attendance.js

```diff
@@ -0,0 +1,131 @@
+import { LightningElement, track, wire, api } from "lwc";
+import { handleError, getChildObjectByName } from "c/util";
+import { getRecord } from "lightning/uiRecordApi";
+import generateRoster from "@salesforce/apex/AttendanceController.generateRoster";
+import getFieldSet from "@salesforce/apex/FieldSetController.getFieldSetForLWC";
+
+import SERVICEDELIVERY_OBJECT from "@salesforce/schema/ServiceDelivery__c";
+import CONTACT_FIELD from "@salesforce/schema/ServiceDelivery__c.Contact__c";
+import QUANTITY_FIELD from "@salesforce/schema/ServiceDelivery__c.Quantity__c";
+import UNITMEASUREMENT_RELATED_FIELD from "@salesforce/schema/ServiceSession__c.ServiceSchedule__r.Service__r.UnitOfMeasurement__c";
+import UNITMEASUREMENT_SERVICE_FIELD from "@salesforce/schema/Service__c.UnitOfMeasurement__c";
+import ATTENDANCE_STATUS_FIELD from "@salesforce/schema/ServiceDelivery__c.AttendanceStatus__c";
+import CREATED_DATE_FIELD from "@salesforce/schema/ServiceDelivery__c.CreatedDate";
+import CREATED_BY_FIELD from "@salesforce/schema/ServiceDelivery__c.CreatedById";
+
+import Submit_Label from "@salesforce/label/c.Submit";
+import Attendance_Label from "@salesforce/label/c.Attendance";
+
+const FIELD_SET_NAME = "Attendance_Service_Deliveries";
+const SHORT_DATA_TYPES = ["DOUBLE", "INTEGER", "BOOLEAN"];
+const LONG_DATA_TYPES = ["TEXTAREA", "PICKLIST", "REFERENCE"];
+
+export default class Attendance extends LightningElement {
+    @api recordId;
+    @track serviceDeliveries;
+    @track fieldSet;
+
+    unitOfMeasurement;
+
+    labels = {
+        submit: Submit_Label,
+        attendance: Attendance_Label,
+    };
+
+    fields = {
+        contact: CONTACT_FIELD,
+        quantity: QUANTITY_FIELD,
+        attendanceStatus: ATTENDANCE_STATUS_FIELD,
+        createdDate: CREATED_DATE_FIELD,
+        createdBy: CREATED_BY_FIELD,
+    };
+
+    @wire(getRecord, {
+        recordId: "$recordId",
+        fields: [UNITMEASUREMENT_RELATED_FIELD],
+    })
+    wiredSession(result) {
+        if (!(result.data || result.error)) {
+            return;
+        }
+
+        if (result.data) {
+            let schedule = getChildObjectByName(result.data.fields, "ServiceSchedule__r");
+            let service = getChildObjectByName(schedule.value.fields, "Service__r");
+            this.unitOfMeasurement =
+                service.value.fields[UNITMEASUREMENT_SERVICE_FIELD.fieldApiName].value;
+        } else if (result.error) {
+            handleError(result.error);
```

Comment:
Can we console.log this here, otherwise when an error occurs they will get a toast even if they haven't interacted with our component. On that note if they remove the contact from the field set an error occurs, do we want to manually require that field. Also we could run into that situation if the user doesn't have access to that field.